content/faq/docker/docker-machine.md

```diff
@@ -0,0 +1,104 @@
+--driver amazonec2 \
+--amazonec2-region ap-southeast-1 \
+--amazonec2-open-port 27017 \
+--amazonec2-security-group security-my-awesome-ec2-project \
+--amazonec2-vpc-id \
+--amazonec2-subnet-id subnet-1 \
+--amazonec2-subnet-id subnet-2 \
+--amazonec2-private-address-only \
+--amazonec2-zone b \
+--amazonec2-zone c \
+--amazonec2-instance-type t3a.medium
+```
+
+At this point, you might be a bit confused, So, lets me explain those arguments to you line by line.
+
+Yeah, you may already know what's this line does.
+
+```sh
+docker-machine create my-awesome-ec2-project \
+--driver amazonec2 \
+```
+
+This line allowed you to create a new instance Ubuntu machine, describing which region to settle down.
+
+By the way, If you've already configured the AWS Access Key and Secret Key for using at the AWS CLI's local machine. The Docker Machine command will bring on those kinds of secret keys from your system automatically.
+
+```sh
+--amazonec2-open-port 27017 \
+--amazonec2-open-port 80 \
+```
+
+
+The definition of this parameter opens port 27017 and port 80. In addition, these arguments set the security group for HTTP on our ec2 instance.
+
+Next, What about EC2 networking between the cluster. How argument pattern should we set? OK, take a look.
+
+```sh
+--amazonec2-security-group ${YOUR_SECURITY_GROUP_NAME} \
+--amazonec2-vpc-id ${YOUR_EXIST_VPC_ID} \
+--amazonec2-subnet-id ${YOUR_SUBNET_ID} \
+--amazonec2-private-address-only \
+--amazonec2-zone b \
+--amazonec2-zone c \
+--amazonec2-region ap-southeast-1 \
+```
+
+Which argument means, You specify the security group with your name. Apply the VPC networking and assign a private IP address in a range of the SubnetId. After that, deploy the EC2 instance into the B and C zone around your specific region (ap-southeast-1).
+
+```sh
+--amazonec2-instance-type t3a.medium
+```
+
+This argument specifies the instance type of Docker Machine; It will create AMD CPU type of machine. For example, this instance has chosen the EC2's spec type in t3a.medium (Cost that you should pay for EC2 instance at least 29.00 USD/month).
+
+These are arguments. You can tune up your instance Docker Machine. as such depends on your budget that you are comfortable paying for a project.
+
+After that, you execute the command, assume that you haven't received any error issues while creating the machine. Lastly, Typing a command for reaching the host already build and run onto the AWS EC2.
+
+Example
+
+```sh
+$ docker-machine ls -t 12
+NAME                          ACTIVE   DRIVER      STATE     URL                     SWARM   DOCKER     ERRORS
+my-awesome-project   -        amazonec2   Running   tcp://127.0.0.1:2376           v20.10.5
+```
+
+Finally, It's time to connect and access for playing around with your machine. You can use the command eval that makes a secure connection between your local computer and the virtual server onto AWS EC2.
+
+Example
+```sh
+$ eval $(docker-machine env my-awesome-project)
```

Comment:
I haven't used docker-machine before, does `eval $(docker-machine env my-awesome-project)` make `my-awesome-project` active machine?

If so, can we issue a normal Docker command like `docker run` and will it be sent to my-awesome-project in EC2 instance?

If my assumption is correct, this is an interesting feature of EC2 docker-machine.
Therefore, please add some contents that we can use a normal docker command and how to switch back to local docker on your local machine.

Reply:
It will send to the instance for sure, but I would like him to clarify about the docker machine parameters

Reply:
@aaronamm, The answer is Yes. You always use docker's command as you wish.  In addition, If you've connected with your remote machine. You could interact with your Terminal that will send and execute commands onto your remote instance.